### PR TITLE
Docs: Add call to action for early adopters to README and landing page [ENG-186]

### PR DESCRIPTION
**Pull request by AdrianoKF**
This PR adds a call to action to early adopters to the documentation landing page and (repository) README.

https://github.com/aai-institute/lakefs-spec/security/dependabot/8 is addressed by updating the documentation dependencies.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,10 @@ Highlights:
 - Zero-config option through config autodiscovery
 - Automatic up-/download management to avoid unnecessary transfers for unchanged files
 
+> [!NOTE]
+> lakeFS-spec is a young project - we are looking for early adopters!
```
[the remark this answers is not in the document]
Comment (AdrianoKF) [reply README.md]: IMO, that deviates from the original spirit of the story, but I like the wording. Slightly adapted it in 79d6b37.

File: docs/index.md [hunk 2]
```diff
@@ -1,6 +1,11 @@
 # lakefs-spec: An fsspec backend for lakeFS
 
-Welcome to lakeFS-spec, a [filesystem-spec](https://github.com/fsspec/filesystem_spec) backend implementation for the [lakeFS](https://lakefs.io/) data lake.
+!!! tip "Early Adopters"
+    lakeFS-spec is a young project - we are looking for early adopters!
```
[the remark this answers is not in the document]
Comment (AdrianoKF) [reply docs/index.md]: Reworded in 79d6b37.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Maciej818) [reply README.md]: I like https://github.com/aai-institute/lakefs-spec/commit/79d6b37a5aa2c8bf961167b6d86d262666f085fe .

File: docs/index.md [hunk 3]
```diff
@@ -1,6 +1,10 @@
 # lakefs-spec: An fsspec backend for lakeFS
 
-Welcome to lakeFS-spec, a [filesystem-spec](https://github.com/fsspec/filesystem_spec) backend implementation for the [lakeFS](https://lakefs.io/) data lake.
+!!! tip "Early Adopters"
+    We are seeking early adopters who would like to actively participate in our feedback process and shape the future of the library.
```
Comment (maxmynter) [inline docs/index.md:4]: WDYT about "We are seeking early adopters who we invite  to actively [...]"?

To not scare early adopters off by demanding too much work from them. On the other hand, this makes clear that we are available for help and their feedback can shape the tool.

File: README.md [hunk 4]
```diff
@@ -14,6 +14,10 @@ Highlights:
 - Zero-config option through config autodiscovery
 - Automatic up-/download management to avoid unnecessary transfers for unchanged files
 
+> [!NOTE]
+> We are seeking early adopters who would like to actively participate in our feedback process and shape the future of the library.
```
Comment (maxmynter) [inline README.md:18]: Same here

**Review by maxmynter — APPROVED** [review]

**Pull request merged**